pkg/service/init.go

```diff
@@ -63,95 +121,69 @@ func (a *AusfApp) SetLogLevel(level string) {
 
-	pemPath := factory.AusfDefaultCertPemPath
-	keyPath := factory.AusfDefaultPrivateKeyPath
-	sbi := factory.AusfConfig.Configuration.Sbi
-	if sbi.Tls != nil {
-		pemPath = sbi.Tls.Pem
-		keyPath = sbi.Tls.Key
-	}
-
-	self := a.ausfCtx
-	// Register to NRF
-	profile, err := consumer.BuildNFInstance(self)
-	if err != nil {
-		logger.InitLog.Error("Build AUSF Profile Error")
-	}
-	_, self.NfId, err = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, profile)
-	if err != nil {
-		logger.InitLog.Errorf("AUSF register to NRF Error[%s]", err.Error())
+	if err := a.sbiServer.Run(context.Background(), &a.wg); err != nil {
+		logger.MainLog.Fatalf("Run SBI server failed: %+v", err)
 	}
+}
 
-	addr := fmt.Sprintf("%s:%d", self.BindingIPv4, self.SBIPort)
-
-	signalChannel := make(chan os.Signal, 1)
-	signal.Notify(signalChannel, os.Interrupt, syscall.SIGTERM)
-	go func() {
-		defer func() {
-			if p := recover(); p != nil {
-				// Print stack for panic to log. Fatalf() will let program exit.
-				logger.InitLog.Fatalf("panic: %v\n%s", p, string(debug.Stack()))
-			}
-		}()
-
-		<-signalChannel
-		a.Terminate()
-		os.Exit(0)
+func (a *AusfApp) listenShutdownEvent() {
+	defer func() {
+		if p := recover(); p != nil {
+			// Print stack for panic to log. Fatalf() will let program exit.
+			logger.MainLog.Fatalf("panic: %v\n%s", p, string(debug.Stack()))
+		}
+		a.wg.Done()
 	}()
 
-	server, err := httpwrapper.NewHttp2Server(addr, tlsKeyLogPath, router)
-	if server == nil {
-		logger.InitLog.Errorf("Initialize HTTP server failed: %+v", err)
-		return
-	}
-
-	if err != nil {
-		logger.InitLog.Warnf("Initialize HTTP server: +%v", err)
-	}
-
-	serverScheme := factory.AusfConfig.Configuration.Sbi.Scheme
-	if serverScheme == "http" {
-		err = server.ListenAndServe()
-	} else if serverScheme == "https" {
-		err = server.ListenAndServeTLS(pemPath, keyPath)
-	}
-
-	if err != nil {
-		logger.InitLog.Fatalf("HTTP server setup failed: %+v", err)
-	}
+	<-a.ctx.Done()
+	a.Terminate()
 }
 
-func (a *AusfApp) Terminate() {
-	logger.InitLog.Infof("Terminating AUSF...")
+func (c *AusfApp) Terminate() {
+	logger.MainLog.Infof("Terminating AUSF...")
+	c.cancel()
+	c.CallServerStop()
+
 	// deregister with NRF
-	problemDetails, err := consumer.SendDeregisterNFInstance()
+	problemDetails, err := c.Consumer().SendDeregisterNFInstance()
 	if problemDetails != nil {
-		logger.InitLog.Errorf("Deregister NF instance Failed Problem[%+v]", problemDetails)
+		logger.MainLog.Errorf("Deregister NF instance Failed Problem[%+v]", problemDetails)
 	} else if err != nil {
-		logger.InitLog.Errorf("Deregister NF instance Error[%+v]", err)
+		logger.MainLog.Errorf("Deregister NF instance Error[%+v]", err)
 	} else {
-		logger.InitLog.Infof("Deregister from NRF successfully")
+		logger.MainLog.Infof("Deregister from NRF successfully")
+	}
+	logger.MainLog.Infof("AUSF SBI Server terminated")
+}
+
+func (a *AusfApp) CallServerStop() {
+	if a.sbiServer != nil {
+		a.sbiServer.Stop()
```

Comment:
```suggestion
func (a *AusfApp) CallServerShutdown() {
	if a.sbiServer != nil {
		a.sbiServer.Shutdown()
```

Align naming convention.

Reply:
Hi yccodr. Thanks for your comment. I have made the corrections you suggested. Please help me check it. Thank you.